packages/node_modules/@webex/plugin-meetings/src/stats/events.js

```diff
@@ -0,0 +1,136 @@
+      something.match = {
+        type,
+        key,
+        data: operate[key],
+        stat,
+        kind
+      };
+      return true;
+    }
+    return false;
+  });
+  return something;
+};
+
+export default class StatsEvents extends StatelessSparkPlugin {
+  namespace = MEETINGS;
+
+  /**
+   * constructs an instance
+   * @param {Object} attrs
+   * @param {Object} options
+   */
+  constructor(attrs, options) {
+    super({}, options);
+    this.sentFirstVideoBytes = undefined;
+    this.receivedFirstVideoBytes = undefined;
+    this.sentFirstAudioBytes = undefined;
+    this.receivedFirstAudioBytes = undefined;
+  }
+
+  /**
+   * handles the first audio bytes received
+   * @param {WebRTCData} data the filtered, parsed, converted, transformed, simplified data point to check against
+   * @returns {Object} always whatever the first receiveFirstVideoBytes were, in the past or if it happened now, or undefined if never
+   */
+  receiveEventAudio(data) {
+    if (data && data.data && data.data.getData()) {
+      const operate = data.data.getData();
+      if (operate && !this.receivedFirstAudioBytes) {
+        const somethingMatched = operateEvent(operate, EVENT_TYPES.REMOTE, DEFAULT_EVENT_AUDIO_RECEIVE_KEYS, EVENT_STATS_MAP.BYTES_RECEIVED, VIDEO);
+        if (somethingMatched && somethingMatched.found && somethingMatched.match) {
+          this.receivedFirstAudioBytes = somethingMatched.match;
+          this.trigger(EVENT_TRIGGERS.STATS_UPDATE, this.receivedFirstAudioBytes);
+        }
+      }
+    }
+    return this.receivedFirstVideoBytes;
+  }
+
+  /**
+   * handles the first video bytes received
+   * @param {WebRTCData} data the filtered, parsed, converted, transformed, simplified data point to check against
+   * @returns {Object} always whatever the first receiveFirstVideoBytes were, in the past or if it happened now, or undefined if never
+   */
+  receiveEventVideo(data) {
+    if (data && data.data && data.data.getData()) {
+      const operate = data.data.getData();
+      if (operate && !this.receivedFirstVideoBytes) {
+        const somethingMatched = operateEvent(operate, EVENT_TYPES.REMOTE, DEFAULT_EVENT_VIDEO_RECEIVE_KEYS, EVENT_STATS_MAP.BYTES_RECEIVED, VIDEO);
+        if (somethingMatched && somethingMatched.found && somethingMatched.match) {
+          this.receivedFirstVideoBytes = somethingMatched.match;
+          this.trigger(EVENT_TRIGGERS.STATS_UPDATE, this.receivedFirstVideoBytes);
+        }
+      }
+    }
+    return this.receivedFirstVideoBytes;
+  }
+
+  /**
+   * handles the first video bytes sent
+   * @param {WebRTCData} data the filtered, parsed, converted, transformed, simplified data point to check against
+   * @returns {Object} always whatever the first sentFirstVideoBytes were, in the past or if it happened now, or undefined if never
+   */
+  sendEventVideo(data) {
+    if (data && data.data && data.data.getData()) {
```

Comment:
`data.data.getData()` weeeeee

Reply:
yeah this was something i was going to look at when i dived in to the analysis, because the WebRTCData will be more defined at that point..

Reply:
I will mark it with a TODO, would not be hard to change later on, also has to do with that Transform stream callback

Reply:
marked in the WebRTCData class